src/streaming/store.rs

```diff
@@ -290,3 +291,52 @@ pub fn flush_chunks(
         &client_path,
     )
 }
+
+pub fn partial_read_file(
+    store: impl Store,
+    client_id: &Path,
+    path: &PathBuf,
+    location: Location,
+    offset: usize,
+    length: usize,
+) -> Result<(Message, usize), Error> {
+    if length > MAX_MESSAGE_LENGTH {
+        return Err(Error::FilesystemReadFailure);
+    }
+    let path = actual_path(client_id, path)?;
+    let offset = u32::try_from(offset).map_err(|_| Error::FilesystemReadFailure)?;
+    let pos = OpenSeekFrom::Start(offset);
+    let (mut data, file_length) = read_chunk(store, location, &path, pos)?;
+    data.truncate(length);
```

Comment:
I think it would be better to read directly the correct length.

Reply:
Fixed.